3rd-party/woocommerce-services.php

```diff
@@ -0,0 +1,88 @@
+<?php
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+class WC_Services_Installer {
+
+	/**
+	 * @var WC_Services_Installer
+	 **/
+	private static $instance = null;
+
+	static function init() {
+		if ( is_null( self::$instance ) ) {
+			self::$instance = new WC_Services_Installer();
+		}
+		return self::$instance;
+	}
+
+	public function __construct() {
+		add_action( 'admin_init', array( $this, 'add_error_notice' ) );
+		add_action( 'admin_init', array( $this, 'try_install' ) );
+	}
+
+	public function try_install() {
+		if ( isset( $_GET['wc-services-action'] ) && ( 'install' === $_GET['wc-services-action'] ) ) {
+			check_admin_referer( 'wc-services-install' );
+
+			if ( ! current_user_can( 'install_plugins' ) ) {
+				return;
+			}
+
+			$result   = $this->install();
+			$redirect = wp_get_referer();
+
+			if ( false === $result ) {
```

Comment:
May be easier to just do `if ( ! $this->install() ) {` -- no need for the variable to store it for a couple lines.